tsl/test/isolation/specs/freeze_chunk.spec

```
@@ -57,32 +69,64 @@ step "sel_s2_query"  { BEGIN; SELECT * FROM measurements ORDER BY 1; }
 step "sel_s2_wp_release"      { SELECT debug_waitpoint_release('freeze_chunk_before_lock'); }
 
 session "LC"
+step "BeginLock" {
+	BEGIN;
+}
+
 step "LockChunk1" {
-  BEGIN;
   SELECT
     lock_chunktable(format('%I.%I',ch.schema_name, ch.table_name))
   FROM _timescaledb_catalog.hypertable ht, _timescaledb_catalog.chunk ch
   WHERE ch.hypertable_id = ht.id AND ht.table_name like 'measurements'
   ORDER BY ch.id LIMIT 1;
 }
-step "UnlockChunk" {ROLLBACK;}
+
+step "UpdateCatalogTupleStatus" {
+  UPDATE _timescaledb_catalog.chunk SET status = 4 WHERE true;
+}
+step "Unlock" {ROLLBACK;}
+step "CommitUpdate" {COMMIT;}
 
 session "comp_s2"
 step "comp_s2_compress" 
 	{ BEGIN; SELECT CASE WHEN compress_chunk(ch) IS NOT NULL THEN 'Success' ELSE 'Failed' END  as COL FROM (SELECT show_chunks('measurements') ch ) q; }
 step "comp_s2_commit" { COMMIT; }
 
-###freeze_chunk waits for pending insert txns to complete
+session "decomp_s2"
+step "decomp_s2_decompress" 
+	{ BEGIN; SELECT CASE WHEN decompress_chunk(ch) IS NOT NULL THEN 'Success' ELSE 'Failed' END  as COL FROM (SELECT show_chunks('measurements') ch ) q; }
+step "decomp_s2_commit" { COMMIT; }
+
+#### freeze_chunk ####
+
+###freeze_chunk waits for pending insert txns to complete, uncompressed and compressed variant
 permutation "s2_wp_enable" "ins_s2_insert" "s2_wp_release" "s1_freeze" "ins_s2_commit" "ins_s2_query" "s1_status"
+permutation "comp_s2_compress" "comp_s2_commit" "s2_wp_enable" "ins_s2_insert" "s2_wp_release" "s1_freeze" "ins_s2_commit" "ins_s2_query" "s1_status"
 
-###freeze_chunk waits for pending update txns to complete
+###freeze_chunk waits for pending update txns to complete, uncompressed and compressed variant
```

Could you please add a comment so that this is clear for future eng who touch this file.
```suggestion
The transactions are serialized when they try to lock the chunk tuple entry in the catalog table. (See chunk.c)
```